silkworm/core/protocol/validation.cpp

```diff
@@ -187,6 +199,12 @@ ValidationResult validate_call_precheck(const Transaction& txn, const EVM& evm)
         }
     }
 
+    if (evm.revision() >= EVMC_PRAGUE) {
+        if (txn.type == TransactionType::kSetCode && !txn.to) {
```

Comment:
Hmm. I'm not sure what's the difference between `validate_call_precheck` & `pre_validate_transaction` is (looks like we should eliminate code duplication between the two methods), but it looks suspicious to me that EIP-7702 transactions are validated differently between the two.

Reply:
`pre_validate_transaction` and `validate_call_precheck` are slightly different because the checks done also by Erigon before call simulations (e.g. `eth_call`) are somewhat more relaxed.
Anyway, validation of EIP-7702 transactions seems to be equivalent except for the missing check on Prague revision in `pre_validate_transaction`, which should be added.

I agree we should eliminate duplication between the two methods (see #2522), doing it in separate PR is fine or even better.